llama_index/tools/types.py

```diff
@@ -50,6 +54,27 @@ def to_openai_function(self) -> Dict[str, Any]:
             "parameters": parameters,
         }
 
+    def to_openai_tool(self) -> Dict[str, Any]:
```

Comment:
is this just duplicate code with `to_openai_function`? Can we just have to_openai_function call `to_openai_tool` while still marking it as deprecated?

Reply:
Yeah you're right. I made the change as such. Though if a user uses `to_openai_function` and tries to pass it into `llm.chat()` as follows, then it won't work as I made the change on that method.

```
llm.chat(..., tool_calls=[...], tool_choice="auto")
```
(i.e., no longer `function_call` or `functions` in the signature)

Reply:
I just noticed that you got it handled in `openai_assistant_agent.py` thru the required `{'type': ... 'function': ...}`.

Reply:
i think i see what you're saying

Reply:
very nit: Don't we want to move the impl to `to_openai_tool` and have `to_openai_function` call it and marked as deprecated?

Reply:
Yeah, it's subtle but `to_openai_tool` is not just a name change from `to_openai_function`. The former wraps the output of the latter in an outer dict. So, I was worried that the user might get confused if they for some reason wanted the original behaviour of `to_openai_function`. 

But I can't think of any cases where a user would want to get that, so I feel what you both are suggesting is the most sensible approach. 

If a user uses `to_openai_function` and it makes a queries an llm directly to use function call as below, then it should work still.

```python
openai_function = my_custom_tool.to_openai_function()
OpenAI().complete("Complete this text", tools=[openai_function])
```

The above wouldn't work (I believe OpenAI i think will throw back a bad request on the user) if they instead use `functions` as the argument instead of `tools` when invoking `complete`.